libwacom/libwacom-database.c

```diff
@@ -333,47 +489,51 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 		}
 		g_clear_pointer(&string_list, g_strfreev);
 
-		stylus->has_lens = g_key_file_get_boolean(keyfile, groups[i], "HasLens", &error);
+		stylus->has_lens = boolean_or_fallback(keyfile, groups[i], "HasLens", aliased ? aliased->has_lens : FALSE, stylus, error);
 		if (error && error->code == G_KEY_FILE_ERROR_INVALID_VALUE)
 			g_warning ("Stylus %s (%s) %s\n", stylus->name, groups[i], error->message);
 		g_clear_error (&error);
-		stylus->has_wheel = g_key_file_get_boolean(keyfile, groups[i], "HasWheel", &error);
+		stylus->has_wheel = boolean_or_fallback(keyfile, groups[i], "HasWheel", aliased ? aliased->has_wheel : FALSE, stylus, error);
 		if (error && error->code == G_KEY_FILE_ERROR_INVALID_VALUE)
 			g_warning ("Stylus %s (%s) %s\n", stylus->name, groups[i], error->message);
 		g_clear_error (&error);
-
-		stylus->num_buttons = g_key_file_get_integer(keyfile, groups[i], "Buttons", &error);
+		stylus->num_buttons = int_or_fallback(keyfile, groups[i], "Buttons", aliased ? aliased->num_buttons : 0, error);
```

Comment:
I keep thinking that for the alias is null case:
```
WacomStylus empty = {0};
WacomStylus *aliased = &empty;
```
then you don't need all the `aliased ? aliased->foo : 0` conditions.